### ISSUE TITLE
Research using relative path everywhere for `tar` in `save_cache`

### ISSUE THREAD

**Issue by spencertransier** [Automattic/bash-cache-buildkite-plugin#45]
When adding the `install_swiftpm_dependencies` action, I added a `tar` option in the `save_cache` action that uses a relative path. See this comment thread for details: https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/43#discussion_r1122968427

Existing tar usage: `tar -czf "$CACHE_KEY" "$CACHE_FILE"`
Relative path: `tar -czf "$CACHE_KEY" -C "$CACHE_FILE" .`

Let's research whether the relative path example could be used by all the instances of `save_cache` in the A8C code base.

**Comment by AliSoftware** [comment]
Encountered this issue when I worked on fixing `publish_pod` to add support for `--synchronous` when publishing co-dependent podspecs from a single repo. See https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/85

TL;DR: `pod trunk push --synchronous` relies on using CocoaPods' master spec repo instead of the CDN (to get rid of CDN propagation latencies), and the master repo being very large I tried to use `cache_cocoapods_specs_repos` action for such case. But it ended up [**failing** to restore any file from the cache tarball](https://buildkite.com/automattic/gravatar-sdk-ios/builds/424#018eaa6b-9040-43e8-8119-0361a843b483/6-8) seemingly to that exact relative-path vs absolute-path issue.

It's very possible that using `-C` during `cache_cocoapods_specs_repos` might help solve this issue. I don't have the bandwidth to try rn though, but definitively worth checking at some point.
